### DISCUSSION TITLE
github: 'silent' broke?

### DISCUSSION THREAD

**Discussion by paulirish** [vercel/vercel#9014] [Help] [answered]
Looks like this broke between 11AM (pacific) and 3pm today..

Example PR and comment: https://github.com/GoogleChrome/lighthouse/pull/14587#issuecomment-1338301853
The now.json in that branch: https://github.com/GoogleChrome/lighthouse/blob/report-emulationstr-ifdisabled/now.json

I remember encountering a similar problem and noted that upgrading from the (very old) `now.json` to `vercel.json` resolved it so i tried that....

https://github.com/GoogleChrome/lighthouse/pull/14588  but no dice. Still get a bot comment on every pr (and edit on every push)

**Comment by amyegan** [comment]
Thanks for reaching out Paul! You were right about updating to `vercel.json`, which should have solved the problem. I found one similar report of the `github.silent` setting not working, so I'm asking the team about it and I'll let you know as soon as I learn more.

**Reply by jonioni** [reply]
Encountering same issue since yesterday.

**Reply by alexbouchardd** [reply]
Same here, `github.silent` is no longer respected since yesterday

**Reply by pvdz** [reply]
@jonioni or @alexbouchardd Can you confirm whether this is still the case now?

**Reply by alexbouchardd** [reply]
@pvdz Looks good 👍

**Reply by alexbouchardd** [reply]
@pvdz It seemed to have broke again this morning for us

**Reply by amyegan** [reply]
@alexbouchardd  Do you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can share with the engineering team so we can get a better look at what is going wrong?

**Reply by alexbouchardd** [reply]
I don't and the issue seems sporadic, we just had a few PR's this morning and also get the Vercel comments

**Reply by jonioni** [reply]
The issue was dismissed for a while but seems to be happening again.

**Comment by pvdz** [comment]
Hi @paulirish! :) Looks like I may have broken it in some cases. Taking a look now.

**Reply by pvdz** [reply]
@paulirish can you confirm this is still broken now? I don't see the comment anymore and it looks like it was resolved, just want to confirm it's not some edge case that's somehow not covered.

**Reply by paulirish** [reply]
oh hey dude :) 👋 

> can you confirm this is still broken now? 

Hmm I'm not sure. We had more PRs yesterday that didnt get a comment. But then this older PR got a new one: https://github.com/GoogleChrome/lighthouse/pull/14576#issuecomment-1339916520

note that i did land https://github.com/GoogleChrome/lighthouse/pull/14588 today

and here's a PR after that landing: https://github.com/GoogleChrome/lighthouse/pull/14596 looks good..

--------

also FYI yesterday in the project settings i did switch our primary branch from `master` to `main`. we did the actual branch change a couple months ago.

shrug.  you can be sure i'll holler if this comes back. :)
thanks for your help

**Reply by avarun42** [reply]
still broken for us. We have a `vercel.json` with the property:
```json
"github": {
		"silent": true
	},
```

and it's not being respected as of last week

**Reply by amyegan** [reply]
@avarun42 Do you have a public repo or [MRE](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can share with the engineering team?

**Reply by e-im** [reply]
A not too large public repo experiencing this issue is here: https://github.com/PaperMC/docs
A MRE is difficult as it seems completely random, I haven't been able to reproduce it reliably.

No changes were made to vercel.json, and no changes made in the vercel dashboard (or, at least none that show up under "activity"). No other changes to the repo have been made such as switching the main branch as others have mentioned above

From that repo, some prs in the last few days do have the comment:
https://github.com/PaperMC/docs/pull/122 (from today)
https://github.com/PaperMC/docs/pull/119

While others don't:
https://github.com/PaperMC/docs/pull/120
https://github.com/PaperMC/docs/pull/121

**Reply by e-im** [reply]
I did get it to happen on a more minimal repository (https://github.com/e-im/vercel-9014/pull/8), but it seems to have just randomly happened on the 8th pull request. I'm not sure how much that can really help. All pull requests made are identical, with 4 coming from branches on the repository and another 4 coming from a fork. I thought that might have something to do with it but doesn't seem to.

**Reply by amyegan** [reply]
Thanks for sharing all of these details @e-im! I shared your message with the rest of the team. 

Intermittent problems are tricky and can take a while to solve. I really appreciate everyone's patience and willingness to work with us to solve this 🙇

**Comment by amyegan** [comment, marked as answer]
The engineering team shipped a fix for this problem yesterday. You should no longer see the intermittent PR comments with `github.silent` enabled. 

If you do see issues with this going forward, please let us know!

---
Update: https://github.com/vercel/vercel/discussions/9014#discussioncomment-7184384

**Reply by saurabhnemade** [reply]
I still see it in case of turborepo monorepo setup with multiple vercel.json. I am looking for removing messages generated by vercel bot `vercel bot deployed to Preview – projectname 2 minutes ago ` from github pr comments as I believe it adds too many messages and developers lose the context while reviewing prs.

**Reply by amyegan** [reply]
Are you able to share your repo or a minimal reproduction? We'll need to know more about project structure and configuration to determine what is causing this problem 🙏

**Reply by saurabhnemade** [reply]
@amyegan I've created a poc at: https://github.com/saurabhnemade/vercel-silent-poc/pull/1
Both apps have `silent: true` in `github` config under `vercel.json` yet bot messages are shown as below. 🙏 

<img width="894" alt="Screenshot 2023-04-03 at 18 36 01" src="https://user-images.githubusercontent.com/17445338/229572749-becfa063-609b-4306-a5d9-74c183cbc484.png">

**Reply by e-im** [reply]
The deploy notifications are something different and afaik was never disabled with `silent`. What this issue was originally opened for was vercel still leaving comments when `silent` was set.

![vercel comment](https://i.imgur.com/kghBWXc.png)

**Comment by arvid220u** [comment]
This seems to have happened again? We have silent = true and have been getting the comments starting yesterday.

**Reply by swarnava** [reply]
Hi, We are investigating this. The `github.silent` property has been deprecated in favor of the new settings in the dashboard, which allow for more fine-grained control over which comments appear on your connected Git repositories. These settings can be found in [the Git section of your project's settings](https://vercel.com/docs/deployments/git/vercel-for-github#silence-github-comments).

**Reply by swarnava** [reply]
We have reverted the change although you should still use Project Settings. We will work on migration plan and deprecate it in near future.
